pipeline/transformer/transformer_mets/src/main/scala/uk/ac/wellcome/platform/transformer/mets/service/MetsTransformerWorkerService.scala

```diff
@@ -1,24 +1,30 @@
 package uk.ac.wellcome.platform.transformer.mets.service
 
 import akka.Done
+import com.amazonaws.services.s3.AmazonS3
 import grizzled.slf4j.Logging
-import uk.ac.wellcome.bigmessaging.{BigMessageSender, EmptyMetadata}
+import uk.ac.wellcome.bigmessaging.BigMessageSender
 import uk.ac.wellcome.json.JsonUtil._
 import uk.ac.wellcome.messaging.sns.{NotificationMessage, SNSConfig}
 import uk.ac.wellcome.messaging.sqs.SQSStream
+import uk.ac.wellcome.mets_adapter.models.MetsData
 import uk.ac.wellcome.models.work.internal.TransformedBaseWork
+import uk.ac.wellcome.platform.transformer.mets.client.AssumeRoleClientProvider
 import uk.ac.wellcome.platform.transformer.mets.parsers.MetsXmlParser
-import uk.ac.wellcome.storage.store.{HybridStoreEntry, VersionedStore}
-import uk.ac.wellcome.storage.{Identified, Version}
+import uk.ac.wellcome.storage.store.VersionedStore
+import uk.ac.wellcome.storage.store.s3.S3TypedStore
+import uk.ac.wellcome.storage.streaming.Codec
+import uk.ac.wellcome.storage.{Identified, ObjectLocation, Version}
 import uk.ac.wellcome.typesafe.Runnable
 
 import scala.concurrent.Future
 
 class MetsTransformerWorkerService(
   msgStream: SQSStream[NotificationMessage],
   messageSender: BigMessageSender[SNSConfig, TransformedBaseWork],
-  store: VersionedStore[String, Int, HybridStoreEntry[String, EmptyMetadata]])
-    extends Runnable
+  adapterStore: VersionedStore[String, Int, MetsData],
+  mestStoreAssumeRoleClient: AssumeRoleClientProvider[AmazonS3]
```

Comment:
In my opinion it would be better for this code to just receive a METS xml store as an argument, rather than receiving the special client and creating the store itself: in this way it is more isolated from implementation details (the fact that it is stored in S3 and needs a special client that assumes the role etc), making the code easier to understand, more easily refactored etc

Reply:
I agree. It's just that, with the credentials being temporary, we need to get a client with every request. But I'm going to extract that bit of functionality away so we don't deal with clients here. 

I wrote the simplest thing that could possibly work because I'm still trying to figure out how to test this thing. Moto, as always, doesn't seem to work

Reply:
Makes sense :+1: